jbmc/src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -392,7 +392,8 @@ java_string_library_preprocesst::process_equals_function_operands(
 /// \param type: a type containing a "data" component
 /// \param symbol_table: symbol table
 /// \return type of the "data" component
-static typet get_data_type(const typet &type, const symbol_tablet &symbol_table)
+static const typet&
```

Comment:
Suggest also taking the symbol on line 401 by reference, as this currently comes worryingly close to returning a reference to an `irept` (the handle structure, not the `dt` within) allocated on the stack, though I think the final `component_type` call avoids this fate.